dialer.go

```diff
@@ -255,6 +268,34 @@ func (i *instrumentedConn) Close() error {
 	return nil
 }
 
+func (d *Dialer) reportOpenConns(ctx context.Context) {
```

Comment:
maybe tangential, but do we need to report this every 30 seconds? Can we just report it when it changes?

Reply:
Yes, definitely. I've verified this works just fine too.